src/Components/Components/src/Routing/Router.cs

```diff
@@ -190,28 +192,31 @@ private void Refresh(bool isNavigationIntercepted)
             }
         }
 
-        private async Task RunOnNavigateAsync(string path)
+        private async ValueTask<bool> RunOnNavigateAsync(string path, Task previousOnNavigate)
         {
             // If this router instance does not provide an OnNavigateAsync parameter
             // then we render the component associated with the route as per usual.
             if (!OnNavigateAsync.HasDelegate)
             {
-                return;
+                return true;
             }
 
             // If we've already invoked a task and stored its CTS, then
-            // cancel the existing task.
-            _onNavigateCts?.Dispose();
+            // cancel that existing CTS.
```

Comment:
In general, this comment adds no value. It simply states what the line below is already doing, I would suggest you update it with the reasons why it is important to know to use Cancel vs Dispose here. Essentially, explain the why instead of the what. Otherwise, simply remove the comment.

Reply:
Good feedback -- I've updated the comment to clarify the intent a little bit more.